### PR TITLE
jobs: add kola-secex test

### PR DESCRIPTION
**Pull request by nikita-dubrovskii**
This will test qemu-secex artifact on s390x bulder machine.

### CODE REVIEW

File: jobs/build-arch.Jenkinsfile [hunk 1]
```diff
@@ -306,6 +306,15 @@ lock(resource: "build-${params.STREAM}-${basearch}") {
             pipeutils.build_artifacts(pipecfg, params.STREAM, basearch)
         }
 
+        // secex specific tests
+        if (shwrapCapture("cosa meta --get-value images.qemu-secex") != "None") {
+            stage("Kola:TestSecex") {
+                // FIXME: Once we have valid HKD for the builder we should add `--qemu-secex-hostkey /path/to/key` switch and enable reboot tests
```
Comment (jschintag) [inline jobs/build-arch.Jenkinsfile:312]: I already added the certificate to the builder:
https://github.com/coreos/fedora-coreos-pipeline/issues/833#issuecomment-1494220283
I think we should implement in the test some option to import all certificates from `/data.secex/hostkeys/HKD-*.crt` by default and only take `--qemu-secex-hostkey /path/to/key` as an override.

File: jobs/build.Jenkinsfile [hunk 2]
```diff
@@ -362,6 +362,15 @@ lock(resource: "build-${params.STREAM}") {
...
 
+        // secex specific tests
+        if (shwrapCapture("cosa meta --get-value images.qemu-secex") != "None") {
+            stage("Kola:TestSecex") {
+                // FIXME: Once we have valid HKD for the builder we should add `--qemu-secex-hostkey /path/to/key` switch and enable reboot tests
+                kola(cosaDir: env.WORKSPACE, arch: basearch, skipBasicScenarios: true, skipSecureBoot: true, skipUpgrade: true,
+                     extraArgs: "--qemu-secex --tag secex")
+            }
+        }
+
```
Comment (jschintag) [inline jobs/build.Jenkinsfile:373]: This should not be necessary, as `build` only runs `x86`

File: jobs/build-arch.Jenkinsfile [hunk 3]
```diff
@@ -306,6 +306,15 @@ lock(resource: "build-${params.STREAM}-${basearch}") {
...
+                kola(cosaDir: env.WORKSPACE, arch: basearch, skipBasicScenarios: true, skipSecureBoot: true, skipUpgrade: true,
+                     extraArgs: "--qemu-secex --tag secex")
```
Comment (jschintag) [inline jobs/build-arch.Jenkinsfile:314]: I'm not that into the implementation of the kola function, but would it not make more sense to implement a setting here like `runOnlySecex: true`, to specify that we only want secex tests to run? This way, with all the skips seems more prone to error when future tests are added.

File: jobs/build.Jenkinsfile (hunk 2 above)
Comment (nikita-dubrovskii) [reply jobs/build.Jenkinsfile:373]: Thx, will drop

File: jobs/build-arch.Jenkinsfile (hunk 1 above)
Comment (nikita-dubrovskii) [reply jobs/build-arch.Jenkinsfile:312]: good to know, so now we can mount them `-v /data.secex/hostkeys/key:/secex-hostkey`

File: jobs/build-arch.Jenkinsfile (hunk 3 above)
Comment (nikita-dubrovskii) [reply jobs/build-arch.Jenkinsfile:314]: maybe, but than i have to update https://github.com/coreos/coreos-ci-lib/blob/main/vars/kola.groovy

Comment (nikita-dubrovskii) [reply jobs/build-arch.Jenkinsfile:314]: @jlebon wdyt?

File: jobs/build-arch.Jenkinsfile (hunk 1 above)
Comment (jschintag) [reply jobs/build-arch.Jenkinsfile:312]: It is already mounted in the CI by default, since it is a volume.

Comment (nikita-dubrovskii) [reply jobs/build-arch.Jenkinsfile]: done

File: jobs/build-arch.Jenkinsfile [hunk 4]
```diff
@@ -306,6 +306,14 @@ lock(resource: "build-${params.STREAM}-${basearch}") {
             pipeutils.build_artifacts(pipecfg, params.STREAM, basearch)
         }
 
+        // secex specific tests
+        if (shwrapCapture("cosa meta --get-value images.qemu-secex") != "None") {
+            stage("Kola:TestSecex") {
```
Comment (jlebon) [inline jobs/build-arch.Jenkinsfile:311]: Minor: maybe just `Kola:Secex`? "Test" is implied by kola. "Kola:TestISO" below is a reference to the command name.

File: jobs/build-arch.Jenkinsfile [hunk 5]
```diff
@@ -306,6 +306,14 @@ lock(resource: "build-${params.STREAM}-${basearch}") {
...
+                kola(cosaDir: env.WORKSPACE, arch: basearch, skipBasicScenarios: true, skipSecureBoot: true, skipUpgrade: true,
```
Comment (jlebon) [inline jobs/build-arch.Jenkinsfile:312]: Passing `extraArgs` already implies that we won't run basic scenarios, so we can simplify this to

```suggestion
                kola(cosaDir: env.WORKSPACE, arch: basearch, skipUpgrade: true,
```

File: jobs/build-arch.Jenkinsfile (hunk 3 above)
Comment (jlebon) [reply jobs/build-arch.Jenkinsfile]: Probably not worth having a helper knob specific to secex. I could imagine in the future though have a `tag` parameter to make this more ergonomic. Anyway, in my review below I mention we can simplify this line today.

**Review by jlebon — APPROVED** [review]
LGTM, though to be safe let's hold it until [the current releases](https://github.com/coreos/fedora-coreos-streams/issues?q=is%3Aissue+2023-04-03) are out.

**Pull request merged**